.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,16 @@
+name: pre-commit
+
+on: [pull_request]
+
+jobs:
+  pre-commit:
+    name: Pre-commit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - run: python -m pip install pre-commit
+      - run: python -m pip freeze --local
+      - run: pre-commit run --all-files
```

Comment:
Pre-commit includes clang-format. We are running clang-format twice then. Better to skip clang-format here with 
run: SKIP=clang-format pre-commit run --all-files.